File: apps/backend/src/organizations/dto/register-organization.input.ts
import { InputType, Field } from '@nestjs/graphql';
import { IsString, IsEmail, IsOptional, MinLength, IsEnum } from 'class-validator';
import { OrganizationType } from '../entities/organization.entity';

@InputType()
export class RegisterOrganizationInput {
  @Field()
  @IsString()
  @MinLength(3)
  name: string;

  @Field(() => String)
  @IsEnum(OrganizationType)
  type: OrganizationType;

  @Field({ nullable: true })
  @IsString()
  @IsOptional()
  legalId?: string;

  @Field({ nullable: true })
  @IsString()
  @IsOptional()
  description?: string;

  @Field({ nullable: true })
  @IsEmail()
  @IsOptional()
  email?: string;

  @Field({ nullable: true })
  @IsString()
  @IsOptional()
  phone?: string;

  @Field({ nullable: true })
  @IsString()
  @IsOptional()
  website?: string;

  @Field({ nullable: true })
  @IsString()
  @IsOptional()
  address?: string;

  @Field({ nullable: true })
  @IsString()
  @IsOptional()
  country?: string;

  @Field({ nullable: true })
  @IsString()
  @IsOptional()
  province?: string;

  @Field({ nullable: true })
  @IsString()
  @IsOptional()
  canton?: string;

  @Field({ nullable: true })
  @IsString()
  @IsOptional()
  district?: string;

  @Field({ nullable: true })
  @IsString()
  @IsOptional()
  legalDocumentationBase64?: string;

  @Field({ nullable: true })
  @IsString()
  @IsOptional()
  financialStatementsBase64?: string;

  @Field({ nullable: true })
  @IsString()
  @IsOptional()
  maxCapacity?: number;
}

File: apps/mobile/lib/features/organizations/data/models/register_organization_input.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:altrupets/features/organizations/data/models/organization.dart';

part 'register_organization_input.freezed.dart';
part 'register_organization_input.g.dart';

@freezed
class RegisterOrganizationInput with _$RegisterOrganizationInput {
  const factory RegisterOrganizationInput({
    required String name,
    required OrganizationType type,
    String? legalId,
    String? description,
    String? email,
    String? phone,
    String? website,
    String? address,
    String? country,
    String? province,
    String? canton,
    String? district,
    String? legalDocumentationBase64,
    String? financialStatementsBase64,
    int? maxCapacity,
  }) = _RegisterOrganizationInput;

  factory RegisterOrganizationInput.fromJson(Map<String, dynamic> json) =>
      _$RegisterOrganizationInputFromJson(json);
}
